store/sqlstore/channel_store.go

```diff
@@ -1563,8 +1564,7 @@ func (s SqlChannelStore) getByName(teamId string, name string, includeDeleted bo
 		return nil, errors.Wrapf(err, "failed to find channel with TeamId=%s and Name=%s", teamId, name)
 	}
 
-	channelByNameCache.SetWithExpiry(teamId+name, &channel, ChannelCacheDuration)
-	return &channel, nil
+	return &channel, channelByNameCache.SetWithExpiry(teamId+name, &channel, ChannelCacheDuration)
```

Comment:
Let's revert this change. This is making it slightly harder to read.

Reply:
Revert it or assign it to `err` and return that?

Reply:
Assign to `err` and return :)